### PR TITLE
web/ui: Add graph view

### PR DESCRIPTION
**Pull request by rfratto**
This PR follows up on #2166 and adds a Graph view which can be used to visualize the relationships between running components:

image

The logic here is the most complicated and probably needs the most attention.

### CODE REVIEW

File: web/ui/src/features/component/ComponentGraph.tsx [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+      });
+
+    svgSelection.call(zoom).call(zoom.transform, d3Zoom.zoomIdentity);
+
+    // Add a marker element so we can draw an arrow pointing between nodes.
+    svgWrapper
+      .append('defs')
+      .append('marker')
+      .attr('id', 'arrow')
+      .attr('viewBox', [0, 0, 20, 20])
```
Comment (mattdurham) [inline web/ui/src/features/component/ComponentGraph.tsx:211]: What are the 0,0,20,20 referenced a few times? Are those the dimensions for the box with the component name?

**Review by rlankfo — APPROVED** [review]
¯\_(ツ)_/¯

File: web/ui/src/features/component/ComponentGraph.tsx (hunk 1 above)
Comment (rfratto) [reply web/ui/src/features/component/ComponentGraph.tsx:211]: Yeah, this assigns a [viewBox attribute](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/viewBox) with x=0, y=0, w=20, h=20. It's basically the viewport size where elements can be drawn.

Do you see 0,0,20,20 anywhere else? I only see the one instance here

Comment (mattdurham) [reply web/ui/src/features/component/ComponentGraph.tsx:211]: Ten or so lines down that I assume is drawing the border

Comment (rfratto) [reply web/ui/src/features/component/ComponentGraph.tsx:211]: Oh, that's slightly different. That's the actual arrow shape. The 0,0,20,20 is the actual box the arrow shape can be drawn in. 


I added some comments to try to clear it up, is it better now?

**Review by mattdurham — APPROVED** [review]
LGTM

**Pull request merged**